code/modules/projectiles/gun.dm

```diff
@@ -50,6 +50,9 @@
 
 	var/safety = TRUE//is safety will be toggled on spawn() or not
 	var/restrict_safety = FALSE //To restrict the users ability to toggle the safety
+	var/dna_compare_samples = FALSE //If DNA-lock installed
```

Comment:
Another set of concerns is that:
1: The dna sampling is done in the gun, which is going to cause issues if someone breaks off the lock and tries to put their own one on.
2: The 'dna' actually uses real_names, instead of dna.

Reply:
Correct it also self-corrects when taken off form the gun